lib/vsc/filesystem/oceanstor.py

```diff
@@ -1811,6 +1894,86 @@ def _get_grace_expiration(grace, usage, soft, hard):
 
         return expired
 
+    def list_namespace_snapshots(self, namespace):
```

Comment:
shouldn't we put an update flag here too?

Reply:
there is no local cache for snapshots, not for namespaces or filesystems/dtrees. I think it is safer for snapshots to always query what exists in the storage.